src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -126,7 +126,7 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 	}
 
 	async provideCompletions(promptValue: string, cursorPosition: number, shellType: TerminalShellType, token: CancellationToken, triggerCharacter?: boolean, skipExtensionCompletions?: boolean): Promise<ITerminalCompletion[] | undefined> {
-		if (!this._providers || !this._providers.values) {
+		if (!this._providers || !this._providers.values || cursorPosition < 0) {
```

Comment:
This shouldn't be possible. If you can reproduce it could you look at how `PromptInputModel` gets a < 0 `cursorIndex`?

Reply:
This reproduces when one enters a subshell, where shell integration does not work. But yes, I'll look into it further

Reply:
https://github.com/microsoft/vscode/blob/dc9707e723a5bc9c72d97f915c9e3f8f8135e9e9/src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts#L149

Reply:
https://github.com/microsoft/vscode/blob/dc9707e723a5bc9c72d97f915c9e3f8f8135e9e9/src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts#L223